pennylane_lightning/lightning_qubit.py

```diff
@@ -127,56 +227,103 @@ def _asarray(arr, dtype=None):
+        """Reset the device"""
+        super().reset()
 
-    def apply(self, operations, rotations=None, **kwargs):
-        # State preparation is currently done in Python
-        if operations:  # make sure operations[0] exists
-            if isinstance(operations[0], QubitStateVector):
-                self._apply_state_vector(operations[0].parameters[0].copy(), operations[0].wires)
-                del operations[0]
-            elif isinstance(operations[0], BasisState):
-                self._apply_basis_state(operations[0].parameters[0], operations[0].wires)
-                del operations[0]
+        # init the state vector to |00..0>
+        self._state = self._create_basis_state(0)
+        self._pre_rotated_state = self._state
 
-        for operation in operations:
-            if isinstance(operation, (QubitStateVector, BasisState)):
-                raise DeviceError(
-                    "Operation {} cannot be used after other Operations have already been "
-                    "applied on a {} device.".format(operation.name, self.short_name)
-                )
+    @property
+    def state(self):
+        # Flattening the state.
+        shape = (1 << self.num_wires,)
+        return self._reshape(self._pre_rotated_state, shape)
 
-        if operations:
-            self._pre_rotated_state = self.apply_lightning(self._state, operations)
-        else:
-            self._pre_rotated_state = self._state
+    def _apply_state_vector(self, state, device_wires):
+        """Initialize the internal state vector in a specified state.
+        Args:
+            state (array[complex]): normalized input state of length ``2**len(wires)``
+                or broadcasted state of shape ``(batch_size, 2**len(wires))``
+            device_wires (Wires): wires that get initialized in the state
+        """
 
-        if rotations:
-            if any(isinstance(r, QubitUnitary) for r in rotations):
-                super().apply(operations=[], rotations=rotations)
-            else:
-                self._state = self.apply_lightning(np.copy(self._pre_rotated_state), rotations)
-        else:
-            self._state = self._pre_rotated_state
+        # translate to wire labels used by device
+        device_wires = self.map_wires(device_wires)
+
+        state = self._asarray(state, dtype=self.C_DTYPE)
+        output_shape = [2] * self.num_wires
+
+        if not qml.math.is_abstract(state):
+            norm = qml.math.linalg.norm(state, axis=-1, ord=2)
+            if not qml.math.allclose(norm, 1.0, atol=tolerance):
+                raise ValueError("Sum of amplitudes-squared does not equal one.")
+
+        if len(device_wires) == self.num_wires and Wires(sorted(device_wires)) == device_wires:
+            # Initialize the entire device state with the input state
+            self._state = self._reshape(state, output_shape)
+            return
+
+        # generate basis states on subset of qubits via the cartesian product
```

Comment:
Hi @AmintorDusko , I'm wondering if operation in batch needs to be supported here?

Reply:
I don't understand. Could you please expand on your question?

Reply:
For example, lines [here](https://github.com/PennyLaneAI/pennylane/blob/37a4024bdf12781dced0c847cdc80edfc321f4b6/pennylane/devices/default_qubit.py#L680-L690)

Reply:
Ah... OK! Following a discussion with @mlxd, since we do not support batching yet, we decide to remove it for now.

Reply:
For context, we need CPP threading support to properly enable this, so we will keep it disabled to allow a safe fallback for now.